### PR TITLE
Remove lsq loop from rebaseline_multi

### PR DESCRIPTION
**Pull request by jakeown**
Here's is my solution to removing the lsq loop that you noticed in previous version.  Now there is only a single lsq call in get_chi. There, I loop through each order of the polynomial, calculate the chisqr value for that model, then select the best model.  Also took care of the numpy to np request, 31 hard-wired value, and sqrt(2) inconsistency.

### CODE REVIEW

File: keystone/gauss_fit.py [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+	GaussNoiseOut = '{0}/{0}_{2}_{1}_gauss_cube_noise.fits'.format(region,root,mol)
+	ParamOut = '{0}/{0}_{2}_{1}_param_cube.fits'.format(region,root,mol)
+
+	# Load the spectral cube and convert to velocity units
+	cube = SpectralCube.read(MolFile)
+	cube_km = cube.with_spectral_unit(u.km / u.s, velocity_convention='radio')
+
+	# If desired, convolve map with larger beam 
+	# or load previously created convolved cube
+	if convolve!=False:
```
Comment (low-sky) [inline keystone/gauss_fit.py]: `if convolve:` ?

File: keystone/gauss_fit.py [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+		cube_km = SpectralCube.read(ConvFile)
+	
+	# Define the spectral axis in km/s
+	spectra_x_axis_kms = np.array(cube_km.spectral_axis) 
+
+	# Find the channel range corresponding to vmin and vmax
+	# -- This is a hold-over from when I originally set up the code to 
+	#    use a channel range rather than velocity range.
+	#    Can change later, but this should work for now. 
+	low_channel = np.where(spectra_x_axis_kms<=vmax)[0][0]+1 # Add ones to change index to channel
```
Comment (low-sky) [inline keystone/gauss_fit.py]: Note also `SpectralCube.closest_spectral_channel()` [docs here](https://spectral-cube.readthedocs.io/en/latest/api/spectral_cube.SpectralCube.html#spectral_cube.SpectralCube.closest_spectral_channel)

File: keystone/gauss_fit.py [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+
+	# Create cubes for storing the fitted Gaussian profiles
+	# and the Gaussians with noise added back into the spectrum
+	header = cube_km.header
+	cube_gauss = np.array(cube_km.unmasked_data[:,:,:])
+	cube_gauss_noise = np.array(cube_km.unmasked_data[:,:,:])
+	shape = np.shape(cube_gauss)
+
+	# Set up a cube for storing fitted parameters
+	param_cube = cube_gauss[0]
```
Comment (low-sky) [inline keystone/gauss_fit.py]: Would this be better as a direct call to `np.zeros()` like `np.zeros(6, shape[1], shape[2])`?

File: keystone/gauss_fit.py [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+	param_cube = np.concatenate((param_cube, param_cube, param_cube, param_cube, param_cube, 	param_cube), axis=0)
+	param_header = cube_km.header
+
+	# Define the Gaussian profile
+	def p_eval(x, a, x0, sigma):
+    		return a*np.exp(-(x-x0)**2/(2*sigma**2))
+
+	# Create some arrays full of NANs
+	# To be used in output cubes if fits fail
+	nan_array=np.empty(shape[0]) # For gauss cubes
```
Comment (low-sky) [inline keystone/gauss_fit.py:112]: Alternatively, initialize to `nan` and then don't overwrite if fail.

File: keystone/gauss_fit.py [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+
+	# Loop through each pixel and find those
+	# with SNR above snr_min
+	x = []
+	y = []
+	pixels = 0
+	for (i,j), value in np.ndenumerate(cube_gauss[0]):
+     		spectra=np.array(cube_km.unmasked_data[:,i,j])
+     		rms = np.std(np.append(spectra[0:(peak_channels[0]-1)], spectra[(peak_channels[1]+1):len(spectra)]))
+     		if (max(spectra[peak_channels[0]:peak_channels[1]]) / rms) > snr_min:
```
Comment (low-sky) [inline keystone/gauss_fit.py]: Are you okay with this failing if there is a `nan` in the array?

File: keystone/gauss_fit.py [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+	# Define a Gaussian fitting function for each pixel
+	# i, j are the x,y coordinates of the pixel being fit
+	def pix_fit(i,j):
+		spectra = np.array(cube_km.unmasked_data[:,i,j])
+		# Use the peak brightness Temp within specified channel 
+		# range as the initial guess for Gaussian height
+		Tpeak = max(spectra[peak_channels[0]:peak_channels[1]])
+		# Use the velocity of the brightness Temp peak as 
+		# initial guess for Gaussian mean
+		vpeak = spectra_x_axis_kms[peak_channels[0]:peak_channels[1]][np.where(spectra[peak_channels[0]:peak_channels[1]]==Tpeak)]
```
Comment (low-sky) [inline keystone/gauss_fit.py]: `np.argmax()`?

File: keystone/gauss_fit.py [hunk 7]
```diff
@@ -0,0 +1,230 @@
...
+		rms = np.std(np.append(spectra[0:(peak_channels[0]-1)], spectra[(peak_channels[1]+1):len(spectra)]))
+		err1 = np.zeros(shape[0])+rms
+		# Create a noise spectrum based on rms of off-line channels
+		# This will be added to best-fit Gaussian to obtain a noisy Gaussian 
+		noise=np.random.uniform(-1.*rms,rms,len(spectra_x_axis_kms))
```
Comment (low-sky) [inline keystone/gauss_fit.py]: I'm unsure as to why you'd want to use uniformly distributed noise here?  Wouldn't you want a normal?

File: keystone/gauss_fit.py [hunk 8]
```diff
@@ -0,0 +1,230 @@
...
+		return i, j, gauss, params, noisy_gauss
+
+	# Parallel computation:
+	nproc = multicore  # maximum number of simultaneous processes desired
+	queue = pprocess.Queue(limit=nproc)
+	calc = queue.manage(pprocess.MakeParallel(pix_fit))
+	tic=time.time()
+	counter = 0
+
+	# Uncomment to see some plots of the fitted spectra
```
Comment (low-sky) [inline keystone/gauss_fit.py]: Plots can be nice to have without reinstalling the code.  You could think about adding a `doPlot=False` keyword and wrap the block in an `if doPlot:` call to get it if desired.

**Review by low-sky — APPROVED** [review]
@jakeown -- All looks fine. I've put in some readability changes.  The only actual issue I flagged is the `uniform` vs `normal` call in the generation of noise.  I'm not sure that you really want `uniform`.  The rest can go in without issue.

File: keystone/gauss_fit.py (hunk 7 above)
Comment (jakeown) [reply keystone/gauss_fit.py]: Yep, thanks for the catch. I will change this in GAS too

File: keystone/gauss_fit.py (hunk 8 above)
Comment (jakeown) [reply keystone/gauss_fit.py]: Yes, I should add that. Those plots were mainly for me during testing but it should still be an option in the function

File: keystone/gauss_fit.py (hunk 6 above)
Comment (jakeown) [reply keystone/gauss_fit.py]: 👍

File: keystone/gauss_fit.py (hunk 1 above)
Comment (jakeown) [reply keystone/gauss_fit.py]: 👍

File: keystone/gauss_fit.py (hunk 3 above)
Comment (jakeown) [reply keystone/gauss_fit.py]: Definitely. Apparently I thought copying and pasting a parameter name six times was the way to go when I wrote this...

File: keystone/gauss_fit.py (hunk 5 above)
Comment (jakeown) [reply keystone/gauss_fit.py]: I've added a line to skip spectra that only contain nans and also changed to `np.nanstd` to catch spectra with some nans.

**Pull request merged**